### ISSUE TITLE
Warning: The frame.append method is deprecated

### ISSUE THREAD

**Issue by psads-git** [milesgranger/gap_statistic#57]
When using the following code

```
from gap_statistic import OptimalK

optimalK = OptimalK(parallel_backend='joblib')
n_clusters = optimalK(X, cluster_array=np.arange(1,10))
print(n_clusters)
```

I get a lot of warnings like the following:

```
/home/x/.virtualenvs/r-reticulate/lib64/python3.11/site-packages/gap_statistic/optimalK.py:137: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  gap_df = gap_df.append(
```

Could you please fix that?

Thanks in advance!

**Comment by milesgranger** [comment]
Hi, thanks for the report.

As [mentioned in the README](https://github.com/milesgranger/gap_statistic#maintenance-mode), I'm not longer going to invest time in this project outside of commenting and/or reviewing/approving PRs. You're in luck though, b/c this warning seems trivial to fix so good opportunity to open a PR to fix it. :)

**Comment by psads-git** [comment]
Thanks, Miles, for your reply.

I have never done a pull request. If you give me the right directions, I believe I can do that.

**Comment by milesgranger** [comment]
I'd be happy to. I'm not sure exactly where you are, but feel free to read up on forking and making a PR (probably tons of videos on YouTube) and I can comment and review when you open one. No pressure at all, happy to help. For this first warning, it pretty much tells us that line 137 needs changing to use the pandas' "concat" instead of "append".

**Comment by psads-git** [comment]
Thanks, Miles. Yes, it seems easy to fix that. The problem is that I cannot find what the Python file to be modified is -- I only see files in rust.

**Comment by psads-git** [comment]
Never mind, Miles: I have just found it!

**Comment by milesgranger** [comment]
Closed by #58
